File: backend/internal/domain/template.go
package domain

const (
	ScopePage     = "page"
	ScopeDocument = "document"
)

const (
	FormatMarkdown = "markdown"
	FormatJSON     = "json"
	FormatCSV      = "csv"
	FormatText     = "text"
)

// Fixed IDs for the two built-in templates seeded by migration 000002.
const (
	TemplateIDCleanMarkdown  = "00000000-0000-0000-0000-000000000001"
	TemplateIDStructuredJSON = "00000000-0000-0000-0000-000000000002"
)

type Template struct {
	ID           string
	OwnerUserID  *string
	Name         string
	DocTypeHint  string
	Scope        string
	Prompt       string
	OutputFormat string
	JSONSchema   *string
	IsBuiltin    bool
}

File: mobile/lib/features/templates/data/models/template.dart
enum TemplateScope { page, document }

enum OutputFormat { markdown, json, csv, text }

TemplateScope templateScopeFromString(Object? s) =>
    s == 'document' ? TemplateScope.document : TemplateScope.page;

OutputFormat outputFormatFromString(Object? s) {
  switch (s) {
    case 'json':
      return OutputFormat.json;
    case 'csv':
      return OutputFormat.csv;
    case 'text':
      return OutputFormat.text;
    default:
      return OutputFormat.markdown;
  }
}

class Template {
  const Template({
    required this.id,
    required this.name,
    required this.scope,
    required this.outputFormat,
    required this.isBuiltin,
    this.docTypeHint,
  });

  final String id;
  final String name;
  final TemplateScope scope;
  final OutputFormat outputFormat;
  final bool isBuiltin;
  final String? docTypeHint;

  factory Template.fromJson(Map<String, dynamic> json) {
    return Template(
      id: json['id'] as String,
      name: json['name'] as String? ?? '',
      scope: templateScopeFromString(json['scope']),
      outputFormat: outputFormatFromString(json['outputFormat']),
      isBuiltin: json['isBuiltin'] as bool? ?? false,
      docTypeHint: json['docTypeHint'] as String?,
    );
  }
}
